### ISSUE TITLE
Async operations

### ISSUE THREAD

**Issue by Paul-DS** [Nexific/dart_ftpclient#14]
Thank you for this nice library.

I see that all your functions are sync. Is there any plan to provide async commands ?

**Comment by virtualmarc** [comment]
Yes, theres a plan to move to async commands, but I've got no timeframe yet.

**Comment by salim-lachdhaf** [comment]
I agree, with @Paul-DS , it will be great if you move functions to async mode, especially download function ...
